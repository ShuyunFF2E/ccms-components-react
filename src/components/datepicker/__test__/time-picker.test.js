import React from 'react';
import { render, mount } from 'enzyme';
import Datepicker  from '../index';
describe('TimePicker', () => {

  it('常规渲染', () => {
    const wrapper = render(<Datepicker.TimePicker />);
    expect(wrapper.find('.timepicker-hour')).toHaveLength(1);
    expect(wrapper.find('.timepicker-minute')).toHaveLength(1);
    expect(wrapper.find('.timepicker-second')).toHaveLength(1);
  });

  it('默认值', () => {
    const onChange = value => {
      expect(value.hour).toBe('00');
      expect(value.minute).toBe('00');
      expect(value.second).toBe('00');
    };
    const wrapper = mount(<Datepicker.TimePicker onChange={onChange}/>);
    wrapper.find('.timepicker-hour').at(0).simulate('change');
  });
});
