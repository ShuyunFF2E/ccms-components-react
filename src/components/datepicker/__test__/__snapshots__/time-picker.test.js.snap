// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearPicker renders correctly 1`] = `
<div
  class="cloud-timepicker "
>
  <input
    class="timepicker-hour"
    maxlength="2"
    value="00"
  />
  <label
    class="colon"
  >
    :
  </label>
  <input
    class="timepicker-minute"
    maxlength="2"
    value="00"
  />
  <label
    class="colon"
  >
    :
  </label>
  <input
    class="timepicker-second"
    maxlength="2"
    value="00"
  />
</div>
`;
